modules/RtcStats/RtcStats.js

```diff
@@ -0,0 +1,138 @@
+import Logger from '@jitsi/logger';
+import rtcstatsInit from '@jitsi/rtcstats/rtcstats';
+import traceInit from '@jitsi/rtcstats/trace-ws';
+
+const logger = Logger.getLogger(__filename);
+
+/**
+ * Filter out RTCPeerConnection that are created by callstats.io.
+ *
+ * @param {*} config - Config object sent to the PC c'tor.
+ * @returns {boolean}
+ */
+function connectionFilter(config) {
+    if (config?.iceServers[0] && config.iceServers[0].urls) {
+        for (const iceUrl of config.iceServers[0].urls) {
+            if (iceUrl.indexOf('callstats.io') >= 0) {
+                return true;
+            }
+        }
+    }
+}
+
+/**
+ * Class the controls the rtcstats node module.
+ */
+class RtcStats {
+    initialized = false;
+    trace = null;
+    eventCallback = null;
+    closeCallback = null;
+    setTrace;
+    traceConfigured = new Promise(r => {
+        this.setTrace = r;
+    });
+    meetingFqn = null;
+    setMeetingF;
+    meetingFqnSet = new Promise(r => {
+        this.setMeetingF = r;
+    });
+
+    /**
+     * Initializes the underlying rtcstats node moduke.
+     */
+    init(options) {
+        const { rtcstatsUseLibJitsi } = options.analytics;
+
+        if (rtcstatsUseLibJitsi) {
+            if (!this.initialized) {
+                this.meetingFqnSet.then(() => {
+                    logger.info(`meetingFqn is set to ${this.meetingFqn}, now initializing rtcstats`);
+
+                    const { rtcstatsEndpoint, rtcstatsUseLegacy, rtcstatsPollInterval, rtcstatsSendSdp }
+                        = options.analytics;
+
+                    const traceOptions = {
+                        endpoint: rtcstatsEndpoint,
+                        meetingFqn: this.meetingFqn,
+                        onCloseCallback: this.handleTraceWSClose.bind(this),
+                        useLegacy: rtcstatsUseLegacy
+                    };
+
+                    const rtcstatsOptions = {
+                        connectionFilter,
+                        pollInterval: rtcstatsPollInterval,
+                        useLegacy: rtcstatsUseLegacy,
+                        sendSdp: rtcstatsSendSdp,
+                        eventCallback: this.handleRtcStatsEvent.bind(this)
+                    };
+
+                    this.trace = traceInit(traceOptions);
```

Comment:
When reconnecting on mobile we need to reset the trace and reinitialize, which happens here, however we don't want to call rtcstatsInit otherwise it will proxy the global webrtc and gum objects again.
In jitsi-meet this is guarded by this check [here](https://github.com/jitsi/jitsi-meet/blob/a3c9630a19e1f78cea7112ecf10117e5d45020c4/react/features/rtcstats/RTCStats.ts#L138C16-L138C21)